### PR TITLE
{geo}[*/2018b] WRF + WPS 4.0.1 (intel/2018b) + 4.0.2 (foss/2018b)

### PR DESCRIPTION
**Pull request by andreas-h**
this depends on ~~https://github.com/easybuilders/easybuild-easyblocks/pull/1563~~

### CODE REVIEW

File: easybuild/easyconfigs/t/tcsh/tcsh-6.20.00-GCCcore-7.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+toolchain = {'name': 'GCCcore', 'version': '7.3.0'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [
+    'ftp://ftp.astron.com/pub/%(namelower)s',
+    'ftp://ftp.astron.com/pub/%(namelower)s/old',
+]
+
+# use same binutils version that was used when building GCC toolchain
+builddependencies = [('binutils', '2.30', '', True)]
```
[the remark this answers is not in the document]
Comment (andreas-h) [reply easybuild/easyconfigs/t/tcsh/tcsh-6.20.00-GCCcore-7.3.0.eb]: okay, just copy-pasted from the gcccore-6.4.0 one here

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2473.golett.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b4ceb3de7b5da312033c0fc8a5466251 for a full test report.

File: easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+toolchain = {'name': 'intel', 'version': '2018b'}
+toolchainopts = {'opt': False}  # don't use agressive optimization, stick to -O2
+
+source_urls = ['https://github.com/wrf-model/WRF/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = [
+    'WRF_parallel_build_fix.patch',
+    'WRF-%(version)s_netCDF-Fortran_separate_path.patch',
+    # 'WRF-%(version)s_known_problems.patch',  # no known problems as of 2018-10-19
```
Comment (akesandgren) [inline easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb]: You can just drop this commented patch.

File: easybuild/easyconfigs/w/WPS/WPS-4.0.1-intel-2018b-dmpar.eb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    '1379cee8029818e7a69376a7c425d1a00801284182dd0f56a58a753703d5fb2b',  # v4.0.1.tar.gz
+    'ca8e931847658b0fe24283dc16a87479c8093571f4bae9a04fa1ab6cad49bcb1',  # WPS-4.0.1_netCDF-Fortran_seperate_path.patch
+    '279fc79909a3eb0370fea0bdf5690e0d5bfb9c5d04350f10ae0bf48680238111',  # WPS-4.0.1_find-wrfdir.patch
+]
+
+buildtype = "dmpar"
+versionsuffix = '-%s' % buildtype
+
+dependencies = [
+    ('WRF', version, '%s' % versionsuffix),
```
Comment (boegel) [inline easybuild/easyconfigs/w/WPS/WPS-4.0.1-intel-2018b-dmpar.eb]: @andreas-h This is a bit silly, the `'%s'` is pointless, should be:

```python
('WRF', version, versionsuffix),

File: easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb]: @andreas-h  +1, please remove this line?

File: easybuild/easyconfigs/w/WPS/WPS-4.0.1-intel-2018b-dmpar.eb (hunk 3 above)
Comment (andreas-h) [reply easybuild/easyconfigs/w/WPS/WPS-4.0.1-intel-2018b-dmpar.eb]: agreed :-)

File: easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb (hunk 2 above)
Comment (andreas-h) [reply easybuild/easyconfigs/w/WRF/WRF-4.0.1-intel-2018b-dmpar.eb]: done.

**Comment by boegel** [comment]
New test reports on top of latest updates to WRF/WPS easyblocks coming up...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
node2076.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/e40dd561cedb1e277b8eb2486d86a8e6 for a full test report.

**Comment by boegel** [comment]
@andreas-h Did this actually work for you? It seems like it gets stuck for me when running the test cases...

**Comment by andreas-h** [comment]
What's this error telling me? That I should be consistent in WRF versions for the same toolchain versions?

**Comment by andreas-h** [comment]
How should we proceed here?

**Comment by boegel** [comment]
@andreas-h Sorry for not getting back to this...

We have a check in place to avoid that we use different "variants" of dependencies within the same toolchain generation (in this case `2018b`). A dependency variant is basically the version + versionsuffix (+ potentially subtoolchain).

In this case, that check doesn't make a whole lot of sense though, since `WRF` and `WPS` should be kept in sync, I think... Is that right?

If so, we can add an exception in the check to filter out multiple versions of `WPS`, as long as they're only used as a dependency to `WRF`.

**Comment by boegel** [comment]
@andreas-h ping?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
node3110.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/805114f5e3c95e529bd4fbb9877e0e75 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
generoso-1 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/7103bc4aa3969080a5c336afe96c36aa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
node3110.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/f6024e8adc2c1df70a39fdda77aebe6f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
node2450.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/6cf74efe36111f3ae2e8d39287ccfc01 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2414.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/98b9b584217b74baed5f961cdc5f8a5a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3120.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/7ccef45bf112062c1b6b134208a11d42 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @andreas-h!

**Pull request merged**